packages/swapper/src/swappers/RelaySwapper/utils/relayTokenToAsset.ts

```diff
@@ -0,0 +1,31 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+
+import { relayTokenToAssetId } from './relayTokenToAssetId'
+import type { RelayToken } from './types'
+
+export const relayTokenToAsset = (
+  relayToken: RelayToken,
+  assets: Partial<Record<AssetId, Asset>>,
+): Asset => {
+  const assetId = relayTokenToAssetId(relayToken)
+  const maybeAsset = assets[assetId]
+  if (maybeAsset) return maybeAsset
+
+  // It shouldn't happen but...
+  // asset not known by shapeshift
+  // create a placeholder asset using the data we have
+  return {
+    assetId,
+    chainId: relayToken.chainId,
+    name: relayToken.name,
+    precision: relayToken.decimals,
+    symbol: relayToken.symbol,
+    color: '#000000',
+    icon: relayToken.metadata.logoURI ?? '',
+    explorer: '',
+    explorerTxLink: '',
+    explorerAddressLink: '',
```

Comment:
We can get this from the ChainId with relevant utils

Reply:
That's useless, this case shouldn't be consumed it's just to make TS happy, we never present a token that the app doesn't know about

Our li.fi implementation is doing the same thing, but I made it for you in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03) ...